docs/manual/src/internals/lifting_and_lowering.md

```diff
@@ -27,27 +71,61 @@ namespace example {
+
+class PythonUserLayer layer
+class RustUserLayer layer
+class ScaffoldingLayer layer
+
+class RustTitle title
+class PythonTitle title
+class ScaffoldingTitle title
+```
+
+Details:
 
-1. The user-provided calling code invokes the `add_to_list` function that is exposed by the
-   UniFFI-generated foreign language bindings, passing `item` as an appropriate language-native
-   integer.
-2. The foreign language bindings ***lower*** each argument to a function call into
-   something that can be passed over the C-style FFI. Since the `item` argument is a plain integer,
+1. UniFFI generates an `add_to_list` function in the foreign language (Python in the example
+   diagram).  In this function the `item` argument and the return type are language-native types.
+2. The generated function ***lowers*** each argument.  Since the `item` argument is a plain integer,
    it is lowered by casting to an `int32_t`.
-3. The foreign language bindings pass the lowered arguments to a C FFI function named
-   like `example_XYZ_add_to_list` that is exposed by the UniFFI-generated Rust scaffolding.
-4. The Rust scaffolding ***lifts*** each argument received over the FFI into a native
-   Rust type. Since `item` is a plain integer it is lifted by casting to a Rust `i32`.
-5. The Rust scaffolding passes the lifted arguments to the user-provided Rust code for
-   the `add_to_list` function, which returns a `Vec<i32>`.
-6. The Rust scaffolding now needs to ***lower*** the return value in order to pass it back
-   to the foreign language code. Since this is a complex data type, it is lowered by serializing
-   the values into a byte buffer and returning the buffer pointer and length from the
-   FFI function.
-7. The foreign language bindings receive the return value and need to ***lift*** it into an
-   appropriate native data type. Since it is a complex data type, it is lifted by deserializing
-   from the returned byte buffer into a language-native list of integers.
+3. The generated Python function passes the lowered arguments to the Rust scaffolding function.
+   This is a `repr(C)` FFI function in Rust library and named `uniffi_fn_add_to_list` in this example.
+4. The Rust scaffolding function ***lifts*** each argument received over the FFI into a native Rust type.
+   Since `item` is a plain integer no conversion is needed.
+5. The Rust scaffolding passes the lifted arguments to the user-defined Rust `add_to_list` function, which then executes normally.
+6. The Rust scaffolding function receives the return value and now needs to ***lower*** the it to pass it back across the FFI.
+   Since this is a complex data type, it is lowered by serializing the values into a byte buffer (`RustBuffer`), which is then returned.
+7. The generated Python function receives the return value, and then ***lifts*** it to a native data type.
+   Since it is a complex data type, it is lifted by deserializing from the RustBuffer a language-native list of integers.
```

Comment:
and kinda ditto here - the point isn't about the complexity of the type, it's about matching the `::FfiType`.